Source/Core/RequestScheduler.js

```diff
@@ -0,0 +1,383 @@
+        var removeCount = 0;
+        var activeLength = activeRequests.length;
+        for (var i = 0; i < activeLength; ++i) {
+            request = activeRequests[i];
+            if (request.cancelled) {
+                // Request was explicitly cancelled
+                cancelRequest(request);
+            }
+            if (request.state !== RequestState.ACTIVE) {
+                // Request is no longer active, remove from array
+                ++removeCount;
+                continue;
+            }
+            if (removeCount > 0) {
+                // Shift back to fill in vacated slots from completed requests
+                activeRequests[i - removeCount] = request;
+            }
+        }
+        activeRequests.length -= removeCount;
+
+        // Resort the heap since priority may have changed. Distance and sse are updated prior to getting here.
+        requestHeap.heapify();
+
+        // Get the number of open slots and fill with the highest priority requests.
+        // Un-throttled requests are automatically added to activeRequests, so activeRequests.length may exceed maximumRequests
+        var openSlots = Math.max(RequestScheduler.maximumRequests - activeRequests.length, 0);
+        var filledSlots = 0;
+        while (filledSlots < openSlots && requestHeap.length > 0) {
+            // Loop until all open slots are filled or the heap becomes empty
+            request = requestHeap.pop();
+            if (request.cancelled) {
+                // Request was explicitly cancelled
+                cancelRequest(request);
+                continue;
+            }
+
+            if (request.throttleByServer && !serverHasOpenSlots(request.server)) {
+                // Open slots are available, but the request is throttled by its server. Cancel and try again later.
+                cancelRequest(request);
+                continue;
+            }
+
+            startRequest(request);
+            ++filledSlots;
+        }
+
+        updateStatistics();
+    };
+
+    /**
+     * Issue a request. If request.throttle is false, the request is sent immediately. Otherwise the request will be
+     * queued and sorted by priority before being sent.
+     *
+     * @param {Request} request The request object.
+     *
+     * @returns {Promise|undefined} A Promise for the requested data, or undefined if this request does not have high enough priority to be issued.
+     */
+    RequestScheduler.request = function(request) {
+        //>>includeStart('debug', pragmas.debug);
+        Check.typeOf.object('request', request);
+        Check.typeOf.string('request.url', request.url);
+        Check.typeOf.func('request.requestFunction', request.requestFunction);
+        //>>includeEnd('debug');
+
+        ++statistics.numberOfAttemptedRequests;
+
+        var isDataOrBlobUri = isDataUri(request.url) || isBlobUri(request.url);
+        if (!isDataOrBlobUri && !defined(request.server)) {
```

Comment:
If it's a data uri or blob, why put it in the queue at all?  You could just move this check up and call `requestFunction` immediately instead of waiting to the end of the frame.  Is there a hidden benefit to threading it through the system?